File: stack/Stack.cs
using System;

namespace Stack {
	public class Stack<T>{
		private int top;
		private int size;
		private T[] stack;
		public Stack(int size){
			this.top = -1;
			this.size = size;
			stack = new T[size];
		}

		public bool isFull(){
			return this.top == this.size - 1;
		}

		public bool isEmpty(){
			return this.top == -1;
		}

		public void push(T n){
			if(!this.isFull()){
				this.top++;
				this.stack[this.top] = n;
				return;
			}

			Console.WriteLine("Stack is full");
		}

		public T pop(){
			if(!this.isEmpty()){
				T n = this.stack[this.top];
				this.top--;
				return n;
			}

			return (T)(object)Int32.MinValue;
		}

		public void printStack(){
			if(this.isEmpty()){
				Console.WriteLine("Queue is Empty");
				return;
			}

			for(int i = 0; i <= this.top; i++){
				Console.Write("{0} ", this.stack[i]);
			}

			Console.WriteLine("");
		}
	}
	public class MainClass{
		public static void Main(){
			Stack<int> stack = new Stack<int>(10);

			int number = 25;

			while(number > 0){
				stack.push(number % 2);
				number /= 2;
			}
			
			while(!stack.isEmpty()){
				Console.Write(stack.pop());
			}
		}
	}
}

File: stack/stack.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

typedef struct Stack{
	int top;
	int size;
	int *node;
} Stack;

Stack *stack_new(int size){
	Stack *stack = (Stack *)malloc(sizeof(Stack));

	stack->size = size;
	stack->top = -1;
	stack->node = (int *)malloc(size * sizeof(int));

	return stack;
}

bool isFull(Stack *stack){
	return stack->top == stack->size  - 1;
}

bool isEmpty(Stack *stack){
	return stack->top == -1;
}

void stack_push(Stack *stack, int n){
	if(isFull(stack)){
		printf("Stack is full\n");
		return;
	}

	stack->top++;
	stack->node[stack->top] = n;
}

void stack_pop(Stack *stack){
	if(isEmpty(stack)){
		printf("Stack is empty\n");
		return;
	}

	stack->top--;
}

void s_print(Stack *stack){
	if(isEmpty(stack)){
		printf("Stack is empty\n");
		return;
	}

	for(int i = 0; i <= stack->top; i++){
		printf("%d ", stack->node[i]);
	}

	printf("\n");
}

int main(){
	Stack *stack = stack_new(10);
	for(int i = 1; i <= 10; i++){
		stack_push(stack, i);
	}
	stack_pop(stack);
	s_print(stack);
	return 0;
}